src/NzbDrone.Core/Download/Clients/FreeboxDownload/FreeboxDownloadProxy.cs

```diff
@@ -0,0 +1,277 @@
+        {
+            var request = BuildRequest(settings).Resource("/downloads/add").Post();
+            request.Headers.ContentType = "application/x-www-form-urlencoded";
+
+            request.AddFormParameter("download_url", System.Web.HttpUtility.UrlPathEncode(url));
+
+            if (!directory.IsNullOrWhiteSpace())
+            {
+                request.AddFormParameter("download_dir", directory);
+            }
+
+            var response = ProcessRequest<FreeboxDownloadTask>(request.Build(), settings);
+
+            SetTorrentSettings(response.Result.Id, addPaused, addFirst, seedRatio, settings);
+
+            return response.Result.Id;
+        }
+
+        public string AddTaskFromFile(string fileName, byte[] fileContent, string directory, bool addPaused, bool addFirst, double? seedRatio, FreeboxDownloadSettings settings)
+        {
+            var request = BuildRequest(settings).Resource("/downloads/add").Post();
+
+            request.AddFormUpload("download_file", fileName, fileContent, "multipart/form-data");
+
+            if (directory.IsNotNullOrWhiteSpace())
+            {
+                request.AddFormParameter("download_dir", directory);
+            }
+
+            var response = ProcessRequest<FreeboxDownloadTask>(request.Build(), settings);
+
+            SetTorrentSettings(response.Result.Id, addPaused, addFirst, seedRatio, settings);
+
+            return response.Result.Id;
+        }
+
+        public void DeleteTask(string id, bool deleteData, FreeboxDownloadSettings settings)
+        {
+            var uri = "/downloads/" + id;
+
+            if (deleteData == true)
+            {
+                uri += "/erase";
+            }
+
+            var request = BuildRequest(settings).Resource(uri).Build();
+
+            request.Method = HttpMethod.Delete;
+
+            ProcessRequest<string>(request, settings);
+        }
+
+        public FreeboxDownloadConfiguration GetDownloadConfiguration(FreeboxDownloadSettings settings)
+        {
+            var request = BuildRequest(settings).Resource("/downloads/config/").Build();
+
+            return ProcessRequest<FreeboxDownloadConfiguration>(request, settings).Result;
+        }
+
+        public List<FreeboxDownloadTask> GetTasks(FreeboxDownloadSettings settings)
+        {
+            var request = BuildRequest(settings).Resource("/downloads/").Build();
+
+            return ProcessRequest<List<FreeboxDownloadTask>>(request, settings).Result;
+        }
+
+        private static string BuildCachedHeaderKey(FreeboxDownloadSettings settings)
+        {
+            return $"{settings.Host}:{settings.AppId}:{settings.AppToken}";
+        }
+
+        private void SetTorrentSettings(string id, bool addPaused, bool addFirst, double? seedRatio, FreeboxDownloadSettings settings)
+        {
+            if (!addPaused && !addFirst && seedRatio == null)
```

Comment:
To avoid issues later it’d be better to return after processing if the dictionary is empty.

Reply:
Done!